proposals/csharp-12.0/collection-expressions.md

```diff
@@ -102,37 +102,28 @@ Collection literals are [target-typed](https://github.com/dotnet/csharplang/blob
 
 A *collection expression conversion* allows a collection expression to be converted to a type.
 
-The following implicit *collection expression conversions* exist from a collection expression:
-
-* To a single dimensional *array type* `T[]` where:
-
-  * For each *element* `Ei` there is an *implicit conversion* to `T`.
-
-* To a *span type* `System.Span<T>` or `System.ReadOnlySpan<T>` where:
-
-  * For each *element* `Ei` there is an *implicit conversion* to `T`.
-
-* To a *type* with a *[create method](#create-methods)* with *parameter type* `System.ReadOnlySpan<T>` where:
-
-  * For each *element* `Ei` there is an *implicit conversion* to `T`.
-
-* To a *struct* or *class type* that implements `System.Collections.Generic.IEnumerable<T>` where:
-
-  * For each *element* `Ei` there is an *implicit conversion* to `T`.
-
-* To a *struct* or *class type* that implements `System.Collections.IEnumerable` and *does not implement* `System.Collections.Generic.IEnumerable<T>`.
-
-* To an *interface type* `System.Collections.Generic.IEnumerable<T>`, `System.Collections.Generic.IReadOnlyCollection<T>`, `System.Collections.Generic.IReadOnlyList<T>`, `System.Collections.Generic.ICollection<T>`, or `System.Collections.Generic.IList<T>` where:
-
-  * For each *element* `Ei` there is an *implicit conversion* to `T`.
+An implicit *collection expression conversion* exists from a collection expression to the following types:
+* A single dimensional *array type* `T[]`
+* A *span type*:
+  * `System.Span<T>`
+  * `System.ReadOnlySpan<T>`
+* A *type* with a *[create method](#create-methods)*
+* A *struct* or *class type* that implements `System.Collections.IEnumerable`
+* An *interface type*:
+  * `System.Collections.Generic.IEnumerable<T>`
+  * `System.Collections.Generic.IReadOnlyCollection<T>`
+  * `System.Collections.Generic.IReadOnlyList<T>`
+  * `System.Collections.Generic.ICollection<T>`
+  * `System.Collections.Generic.IList<T>`
+
+The implicit conversion exists for the type if the following hold:
+* The type has an [*iteration type*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/statements.md#1295-the-foreach-statement) `U` where the iteration type is determined from a `GetEnumerator` instance method or enumerable interface, not from extension methods.
```

Comment:
> is determined from a `GetEnumerator` instance method or enumerable interface, not from extension methods.

I think this was decided exclusively for "CollectionBuilder" types, but it looks like all other types already satisfy this requirement. I am not suggesting any chnages, just want to make sure the more general wording is intentional.